.github/workflows/ci.yml

```diff
@@ -70,6 +70,14 @@ jobs:
           file: coverage.out
           format: golang
 
+      - name: 'Upload junit-xml artifacts'
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: junit-xml--${{github.run_id}}--${{github.run_attempt}}--${{matrix.os}}-${{matrix.go-version}}
+          path: .build/junit-xml
```

Comment:
This coupling (the output dir is hard-coded in both) between the Go build script and the CI yaml isn't ideal, but I'm thinking we can live with it for now. If we want to eliminate it we could have `-junit-output-dir` and `-junit-file-stem` CLI arguments.

Reply:
Why is this not ideal? Is this because we need to make sure we keep these 2 paths in sync?

Reply:
Right, the `ci.yml` is invoking a Go build script which also hard-codes this path, and it would be ideal for it to be hard-coded in one place only. But it's not unusual to have some duplication like that in GitHub Actions job definitions, so I'd say probably OK for now.

Reply:
Should we make a comment here that this should match `junitDir` in the build script, just for added safety?

Reply:
Sure! Done.